src/pages/Home/index.tsx

```diff
@@ -29,197 +25,169 @@ const Container = styled.div`
   padding: 32px;
 `
 
-declare global {
-  interface Window {
-    ethereum: any
-  }
-}
-
-const Home = ({}: { items: any }) => {
-  //Initiation
-  const [activeList, setActiveList] = useState<'Tags' | 'CDN' | 'Tokens'>(
-    'Tags'
+export const ITEMS_PER_PAGE = 20
+
+const Home: React.FC = () => {
+  let [searchParams, setSearchParams] = useSearchParams()
+
+  const searchQueryKeys = useMemo(
+    () => [
+      searchParams.getAll('registry').toString(),
+      searchParams.getAll('status').toString(),
+      searchParams.getAll('disputed').toString(),
+      searchParams.getAll('network').toString(),
+      searchParams.get('text'),
+      searchParams.get('page'),
+      searchParams.get('orderDirection'),
+    ],
+    [searchParams]
   )
 
-  const [items, setItems] = useState([])
-  const [loading, setLoading] = useState(true)
-  const [error, setError] = useState(null)
-
-  //navigation and search
-  const [searchTerm, setSearchTerm] = useState('')
-  const [currentPage, setCurrentPage] = useState(1)
-
-  //for the pop-up to display details and evidence
-  const [isDetailsModalOpen, setIsDetailsModalOpen] = useState(false)
+  const isDetailsModalOpen = useMemo(
+    () => !!searchParams.get('itemdetails'),
+    [searchParams]
+  )
 
-  const [detailsData, setDetailsData] = useState(null)
-  const [evidences, setEvidences] = useState<any[]>([])
-  const [entryStatus, setEntryStatus] = useState('')
-  const [itemId, setItemId] = useState('')
+  const {
+    isLoading: searchLoading,
+    error: searchError,
+    data: searchData,
+  } = useQuery({
+    queryKey: ['fetch', ...searchQueryKeys],
+    queryFn: () => fetchItems(searchParams),
+  })
+  const {
+    isLoading: countsLoading,
+    error: countsError,
+    data: countsData,
+  } = useQuery({
+    queryKey: ['counts', ...searchQueryKeys],
+    queryFn: () => fetchItemCounts(),
+  })
 
-  //contract state management
-  const [curateContractAddress, setCurateContractAddress] = useState('')
-  const [depositParams, setDepositParams] = useState<DepositParamsType>(null)
+  const currentItemCount = useMemo(() => {
```

Comment:
this useMemo variable looks a bit unreadable with all the nested logic etc, maybe refactor it? what GPT4 quickly suggested for example:

```typescript
// Helper function to check if searchParams values are valid
const areSearchParamsValid = (params) => {
    return params.every(param => param.length > 0);
}

// Helper function to check if network has required length
const isNetworkLengthValid = (network) => network.length === 4;

// Extracted getCount function
const getCount = (registry, status, disputed, countsData) => {
    return status.reduce((sum, statusItem) => {
        const isDisputed = disputed.includes('true');
        const countKey = `${registry}.numberOf${statusItem}${isDisputed ? 'Challenged' : ''}`;
        return sum + (disputed.includes('false') ? countsData[registry][countKey] : 0);
    }, 0);
};

const currentItemCount = useMemo(() => {
    const registry = searchParams.getAll('registry');
    const status = searchParams.getAll('status');
    const disputed = searchParams.getAll('disputed');
    const network = searchParams.getAll('network');
    const text = searchParams.get('text');
    const page = searchParams.get('page');

    if (countsLoading || !areSearchParamsValid([registry, status, disputed, network]) || page === null || !countsData) {
        return undefined;
    } else if (!text && isNetworkLengthValid(network)) {
        const count = registry.reduce((sum, regItem) => sum + getCount(regItem, status, disputed, countsData), 0);
        return count;
    } else {
        if (!searchData || searchData.length > ITEMS_PER_PAGE) return null;
        return searchData.length + (Number(page) - 1) * ITEMS_PER_PAGE;
    }
}, [searchParams, countsData, countsLoading, searchData]);
```

Reply:
wontfix, explained to you before but:
- logic is subject to change
- it makes sense to contain it there, instead of "modularizing" it

clean code is not necessarily the answer every time, sometimes its better to keep the ugly contained. reference:
https://overreacted.io/goodbye-clean-code/